server/src/main/java/org/opensearch/common/geo/PointParser.java

```diff
@@ -0,0 +1,227 @@
+/*
+ *  Copyright OpenSearch Contributors
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.common.geo;
+
+import org.opensearch.OpenSearchParseException;
+import org.opensearch.common.xcontent.XContentParser;
+import org.opensearch.geometry.Point;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Locale;
+
+/**
+ * A utility class with a point parser methods supporting different point representation formats
+ *
+ * @opensearch.internal
+ */
+public class PointParser {
```

Comment:
class naming parser has a different meaning in OpenSearch code. I would suggest to rename this to PointUtils as per your RFC.

Reply:
ACK